### PR TITLE
Filter functionality

### PR DESCRIPTION
**Pull request by ScoobyDubeyDo**
Added the functionality
> price slider
sort  radio
rating  radio
types checkbox
- fixed some styles (grid and basic colors)
- added wishlist template

### CODE REVIEW

File: src/context/filter/filter-context.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { createContext, useContext, useReducer } from "react";
+import { useEffect } from "react";
+import { filterReducer, initialState } from "../../reducers";
+import { useProduct } from "../product/product-context";
```
Comment (znjs) [inline src/context/filter/filter-context.jsx:4]: Suggestion: use barrel export if possible

File: src/pages/Products/components/FilterBox/components/TypesCheckbox/TypesCheckbox.jsx [hunk 2]
```diff
@@ -13,30 +16,26 @@ export const TypesCheckbox = ({ categoriesArray }) => {
...
-                                //               },
-                                //           })
-                                // }
+                                onChange={(e) =>
+                                    e.target.checked
+                                        ? dispatch({
+                                              type: "TYPES",
+                                              payload: {
+                                                  typeOfProd: e.target.value,
+                                                  isChecked: true,
```
Comment (rutvikpumak) [inline src/pages/Products/components/FilterBox/components/TypesCheckbox/TypesCheckbox.jsx:28]: You can write a common function for this

**Review by znjs — APPROVED** [review]
Clean code
Good file Structure
Good naming

**Review by rutvikpumak — COMMENTED** [review]
I have suggested a change. Your code looks good and from next time try not to commit a lot of changes in a single pull request.

**Pull request merged**